syncer: some log are confusing

## Enhancement Task
```
[2024/09/25 18:39:35.430 +00:00] [INFO] [client.go:126] ["region syncer start load region"]",
[2024/09/25 18:39:35.430 +00:00] [INFO] [client.go:129] ["region syncer finished load region"] [time-cost=545ns]",
[2024/09/25 18:39:35.448 +00:00] [ERROR] [client.go:174] ["region sync with leader meet error"] [error="[PD:grpc:ErrGRPCRecv]rpc error: code = Unavailable desc = server not started: rpc error: code = Unavailable desc = server not started"]
```

--- a/pkg/syncer/client.go
+++ b/pkg/syncer/client.go
@@ -104,9 +104,10 @@ func (s *RegionSyncer) StartSyncWithLeader(addr string) {
 		log.Info("region syncer start load region")
 		start := time.Now()
 		err := storage.TryLoadRegionsOnce(ctx, regionStorage, bc.CheckAndPutRegion)
-		log.Info("region syncer finished load regions", zap.Duration("time-cost", time.Since(start)))
 		if err != nil {
-			log.Warn("failed to load regions", errs.ZapError(err))
+			log.Warn("region syncer failed to load regions", errs.ZapError(err), zap.Duration("time-cost", time.Since(start)))
+		} else {
+			log.Info("region syncer finished load regions", zap.Duration("time-cost", time.Since(start)))
 		}
 		// establish client.
 		conn := grpcutil.CreateClientConn(ctx, addr, s.tlsConfig,

